### PR TITLE
Four Key DLC

### PR DESCRIPTION
**Pull request by nkohen**
Re-implemented key derivation to be compatible with traditional key discovery as described [here](https://github.com/discreetlogcontracts/dlcspecs/issues/8#issuecomment-586391136)

### CODE REVIEW

File: dlc/src/main/scala/org/bitcoins/dlc/DLCPublicKeys.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      extPubKey
+        .deriveChildPubKey(BIP32Path.fromString(s"m/0/${nextAddressIndex + 3}"))
+        .get
+        .key
+
+    DLCPublicKeys(
+      fundingKey = fundingPubKey,
+      toLocalCETKey = cetToLocalPubKey,
+      toRemoteCETKey = cetToRemotePubKey,
+      finalAddress = Bech32Address(P2WPKHWitnessSPKV0(finalPubKey), RegTest)
```
Comment (benthecarman) [inline dlc/src/main/scala/org/bitcoins/dlc/DLCPublicKeys.scala]: Should make the network a parameter and maybe give it a default val

**Review by benthecarman — APPROVED** [review]

File: dlc/src/main/scala/org/bitcoins/dlc/DLCPublicKeys.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      toLocalCETKey = cetToLocalPubKey,
+      toRemoteCETKey = cetToRemotePubKey,
+      finalAddress = Bech32Address(P2WPKHWitnessSPKV0(finalPubKey), network)
+    )
+  }
+
+  def fromExtPrivKeyAndIndex(
+      extPrivKey: ExtPrivateKey,
+      nextAddressIndex: Int,
+      network: NetworkParameters = RegTest): DLCPublicKeys = {
```
Comment (Christewart) [inline dlc/src/main/scala/org/bitcoins/dlc/DLCPublicKeys.scala]: why hard coded network?

File: dlc/src/main/scala/org/bitcoins/dlc/DLCPublicKeys.scala [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    toLocalCETKey: ECPublicKey,
+    toRemoteCETKey: ECPublicKey,
+    finalAddress: BitcoinAddress)
+
+object DLCPublicKeys {
+
+  def fromExtPubKeyAndIndex(
+      extPubKey: ExtPublicKey,
+      nextAddressIndex: Int,
+      network: NetworkParameters = RegTest): DLCPublicKeys = {
```
Comment (Christewart) [inline dlc/src/main/scala/org/bitcoins/dlc/DLCPublicKeys.scala]: why hard coded network?

File: dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala [hunk 4]
```diff
@@ -883,7 +883,8 @@ case class BinaryOutcomeDLCClient(
     */
   def executeRemoteUnilateralDLC(
       dlcSetup: SetupDLC,
-      publishedCET: Transaction): Future[UnilateralDLCOutcome] = {
+      publishedCET: Transaction,
+      sweepPrivKey: ECPrivateKey): Future[UnilateralDLCOutcome] = {
```
Comment (benthecarman) [inline dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala:887]: Would it be better to pass in a `SpendingInfo` so we can sweep to any addr we want?

Comment (nkohen) [reply dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala:887]: at some point I think that would be nice, would prefer to do it on another PR though as this one is slowly growing in scope :P

**Pull request merged**